File: Productores/productor-python/producer.py
# producer.py
import pika, json, time, os, socket
RABBIT = os.getenv("RABBIT_HOST", "rabbitmq")
CREDENTIALS = pika.PlainCredentials('user','password')
params = pika.ConnectionParameters(host=RABBIT, credentials=CREDENTIALS)

def main():
    connection = pika.BlockingConnection(params)
    channel = connection.channel()
    channel.queue_declare(queue='eventos', durable=True)

    hostname = socket.gethostname()
    for i in range(1, 21):
        msg = {"producer":"python", "host": hostname, "seq": i}
        channel.basic_publish(
            exchange='',
            routing_key='eventos',
            body=json.dumps(msg),
            properties=pika.BasicProperties(delivery_mode=2)
        )
        print(" [x] Sent", msg)
        time.sleep(0.5)
    connection.close()

if __name__ == "__main__":
    main()

File: Productores/productor-node/producer.js
// producer.js
const amqp = require('amqplib');
const RABBIT = process.env.RABBIT_HOST || 'rabbitmq';
const URL = `amqp://user:pass@${RABBIT}`;

(async () => {
  const conn = await amqp.connect(URL);
  const ch = await conn.createChannel();
  await ch.assertQueue('eventos', { durable: true });

  for (let i=1; i<=20; i++){
    const msg = { producer: 'node', seq: i, host: require('os').hostname() };
    ch.sendToQueue('eventos', Buffer.from(JSON.stringify(msg)), { persistent:true });
    console.log(' [x] Sent', msg);
    await new Promise(r => setTimeout(r, 500));
  }
  await ch.close();
  await conn.close();
})();
